packages/cli/workspace-loader/src/loadDependency.ts

```diff
@@ -135,83 +157,98 @@ async function validateVersionedDependencyAndGetDefinition({
+                );
+                return undefined;
+            }
+        }
+        // otherwise, ensure CLI versions are on the same major + minor versions
+        else if (
+            parsedCliVersion.major !== parsedCliVersionOfDependency.major ||
+            parsedCliVersion.minor !== parsedCliVersionOfDependency.minor
+        ) {
             context.failWithoutThrowing(
-                `${dependency.organization}/${dependency.apiName}@${dependency.version} on a higher version of fern. Upgrade this workspace to ${response.body.cliVersion}`
+                `CLI version is ${response.body.cliVersion}. Expected ${parsedCliVersion.major}.${parsedCliVersion.minor}.x (to match current workspace).`
             );
             return undefined;
-        } else if (parsedYamlVersionOfDependency < YAML_SCHEMA_VERSION) {
-            context.failWithoutThrowing(
-                `${dependency.organization}/${dependency.apiName}@${dependency.version} on a lower version of fern. Upgrade it to ${cliVersion}`
-            );
+        }
+
+        // download API
+        context.logger.info("Downloading...");
+        context.logger.debug("Remote URL: " + response.body.definitionS3DownloadUrl);
+
+        await mkdir(pathToDefinition, { recursive: true });
+
+        try {
+            await downloadDependency({
+                s3PreSignedReadUrl: response.body.definitionS3DownloadUrl,
+                absolutePathToLocalOutput: pathToDefinition
+            });
+        } catch (error) {
+            context.failWithoutThrowing("Failed to download API", error);
             return undefined;
         }
-    }
-    // otherwise, ensure CLI versions are on the same major + minor versions
-    else if (
-        parsedCliVersion.major !== parsedCliVersionOfDependency.major ||
-        parsedCliVersion.minor !== parsedCliVersionOfDependency.minor
-    ) {
-        context.failWithoutThrowing(
-            `CLI version is ${response.body.cliVersion}. Expected ${parsedCliVersion.major}.${parsedCliVersion.minor}.x (to match current workspace).`
-        );
-        return undefined;
-    }
 
-    // download API
-    context.logger.info("Downloading...");
-    const pathToDependency = AbsoluteFilePath.of((await tmp.dir()).path);
-    context.logger.debug("Remote URL: " + response.body.definitionS3DownloadUrl);
-    try {
-        await downloadDependency({
-            s3PreSignedReadUrl: response.body.definitionS3DownloadUrl,
-            absolutePathToLocalOutput: pathToDependency
-        });
-    } catch (error) {
-        context.failWithoutThrowing("Failed to download API", error);
-        return undefined;
+        metadata = response.body;
+        await writeFile(pathToMetadata, JSON.stringify(metadata));
+    } else {
+        metadata = JSON.parse((await readFile(pathToMetadata)).toString());
```

Comment:
Is there any validation, similar to the above that needs to be done after reading in the dir? I'd imagine no since it's version specific, so should be used immediately

Reply:
Yup no validation needs to be done